src/ucs/stats/serialization.c

```diff
@@ -303,22 +303,90 @@ ucs_stats_serialize_binary(FILE *stream, ucs_stats_node_t *root,
 }
 
 static ucs_status_t
-ucs_stats_serialize_text_recurs(FILE *stream, ucs_stats_node_t *node,
-                                ucs_stats_children_sel_t sel, unsigned indent)
+ucs_stats_serialize_text_recurs_filtered(FILE *stream,
+                                         ucs_stats_filter_node_t *filter_node,
+                                         unsigned indent)
 {
-    ucs_stats_node_t *child;
+    ucs_stats_filter_node_t *filter_child;
+    ucs_stats_node_t *node;
     unsigned i;
+    int is_sum = ucs_global_opts.stats_format == UCS_STATS_SUMMARY;
+    char *nl = is_sum ? "" : "\n";
+    char *space =  is_sum ? "" : " ";
+    char *left_b = is_sum ? "{" : "";
+    char *rigth_b = is_sum ? "} " : "";
+    node = ucs_list_head(&filter_node->type_list_head,
+                         ucs_stats_node_t,
+                         type_list);
+    if (!node) {
+        ucs_error("no list for node filter");
+        return UCS_OK;
+    }
+    if (!filter_node->ref_count) {
+        return UCS_OK;
+    }
+    node = ucs_list_head(&filter_node->type_list_head,
+                         ucs_stats_node_t,
+                         type_list);
+    if (!node) {
```

Comment:
ucs_list_head is not returning NULL if list is empty, need to check explicitly with ucs_list_is_empty